File: ts/src/budget.ts
/** ~4 chars/token — a common rough heuristic, good enough for a budget
 * gate rather than exact accounting. Same heuristic used across the
 * sibling ActAuth/SkillGarden projects for consistency. */
export function estimateTokens(text: string): number {
  return Math.ceil(text.length / 4)
}

File: py/src/contextclip/budget.py
from __future__ import annotations

import math


def estimate_tokens(text: str) -> int:
    """~4 chars/token — a common rough heuristic, good enough for a
    budget gate rather than exact accounting. Same heuristic used across
    the sibling ActAuth/SkillGarden projects for consistency."""
    return math.ceil(len(text) / 4)
